rmgpy/solver/base.pyx

```diff
@@ -583,14 +762,120 @@ cdef class ReactionSystem(DASx):
+                elif len(surfaceTotalDivAccumNums)>0: 
+                    maxSurfaceAccumReactionIndex = numpy.argmax(surfaceTotalDivAccumNums)
+                    maxSurfaceAccumReaction = surfaceReactions[maxSurfaceAccumReactionIndex]
+                    maxSurfaceDifLnAccumNum = numpy.log(surfaceTotalDivAccumNums[maxSurfaceAccumReactionIndex])
+            else:
+                maxSurfaceAccumReactionIndex = -1
+                maxSurfaceAccumReaction = None
+                maxSurfaceDifLnAccumNum = 0.0
+                
+            #manage surface reaction movement "on the fly"
+            if maxSurfaceAccumReaction and maxSurfaceDifLnAccumNum > toleranceMoveSurfaceReactionToCore:
+                logging.info('Moving reaction {0} from surface to core'.format(maxSurfaceAccumReaction))
+                surfaceReactions.remove(maxSurfaceAccumReaction)
+                surfaceSpecies,surfaceReactions =self.initialize_surface(coreSpecies,coreReactions,surfaceSpecies,surfaceReactions)
+                surfaceSpeciesIndices = self.surfaceSpeciesIndices
+                surfaceReactionIndices = self.surfaceReactionIndices
+                logging.info('Surface has {0} Species and {1} Reactions'.format(len(surfaceSpeciesIndices),len(surfaceReactionIndices)))
+            
+            #surface flux calculations
+            surfaceSpeciesRates = numpy.zeros(len(surfaceSpeciesIndices))
+            surfaceSpeciesProduction = coreSpeciesProductionRates[surfaceSpeciesIndices]
+            surfaceSpeciesConsumption = coreSpeciesConsumptionRates[surfaceSpeciesIndices]
+            for i in xrange(len(surfaceSpeciesIndices)):
+                surfaceSpeciesRates[i] = max(abs(surfaceSpeciesProduction[i]),abs(surfaceSpeciesConsumption[i]))
+                
+            if len(surfaceSpeciesIndices) > 0:
+                maxSurfaceSpeciesIndex = numpy.argmax(surfaceSpeciesRates)
+                maxSurfaceSpecies = surfaceSpecies[maxSurfaceSpeciesIndex]
+                maxSurfaceSpeciesRate = surfaceSpeciesRates[ind]
+            else:
+                maxSurfaceSpeciesIndex = -1
+                maxSurfaceSpecies = None
+                maxSurfaceSpeciesRate = 0.0
             
+            #manage surface species movement "on the fly"
+            if maxSurfaceSpecies and maxSurfaceSpeciesRate/charRate > toleranceMoveSurfaceSpeciesToCore:
+                logging.info('Moving species {0} from surface to core'.format(maxSurfaceSpecies))
+                surfaceSpecies.remove(maxSurfaceSpecies)
+                surfaceSpecies,surfaceReactions = self.initialize_surface(coreSpecies,coreReactions,surfaceSpecies,surfaceReactions)
+                surfaceSpeciesIndices = self.surfaceSpeciesIndices
+                surfaceReactionIndices = self.surfaceReactionIndices
+                logging.info('Surface has {0} Species and {1} Reactions'.format(len(surfaceSpeciesIndices),len(surfaceReactionIndices)))
+                
```

Comment:
That's a good idea, however, I make some pretty sweeping changes to how this is done and eliminate some of the code segments altogether in #1082 so I would prefer to split them up in that pull request instead of this one.